datashuttle/tui/custom_widgets.py

```diff
@@ -39,6 +42,19 @@
 # --------------------------------------------------------------------------------------
 
 
+class ClipboardErrorDialog(ModalDialog):
```

Comment:
This is a nice solution! We have something very similar in `modal_dialogs.MessageBox` and a mainwindow function `show_modal_error_dialog` and so it makes sense to use these if possible, it would be something like:

```
            try:
                pyperclip.copy(self.value)
            except pyperclip.PyperclipException:
                self.mainwindow.show_modal_error_dialog(
                    "Clipboard copy failed, likely due to operating in headless mode (e.g. on a high-performance computer)."
               )
  ```